### ISSUE TITLE
unarchive with copy=no failing inexplicably

### ISSUE THREAD

**Issue by jgrigonis** [ansible/ansible#11274]
I have a playbook to install python.  Works great under 1.8.4. Worked for a long time under 1.9.x.  Fails today on 2.0.0-devel

I threw in the stat code for testing, but it doesn't make it work.

```
- name: configure /usr/src/python permissions
  file: "dest=/usr/src/python state=directory owner={{ ansible_ssh_user }} group={{ ansible_ssh_user }} recurse=yes"
  sudo: yes
- name: get python
  get_url: "url=http://www.python.org/ftp/python/{{ python_version }}/Python-{{ python_version }}.tar.xz dest=/usr/src/python"
- stat: "path=/usr/src/python/Python-{{ python_version }}.tar.xz"
  register: st
- name: unarchive python
  unarchive: "src=/usr/src/python/Python-{{ python_version }}.tar.xz dest=/usr/src/python copy=no"
  when: st.stat.exists == True
```

The error I get:
fatal: [aws-ansible]: FAILED! => {"msg": "ERROR! file or module does not exist: /usr/src/python/Python-2.7.10.tar.xz", "failed": true}

**Comment by bcoca** [comment]
at which task does it fail?could you paste the verbose output?

**Comment by jgrigonis** [comment]
unarchive

TASK [unarchive python] *********************************************************
<aws-ansible> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<aws-ansible> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 aws-ansible mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434403724.27-46406947970554 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1434403724.27-46406947970554 && echo $HOME/.ansible/tmp/ansible-tmp-1434403724.27-46406947970554
<aws-pipeline4> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<aws-pipeline4> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 aws-pipeline4 mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434403724.36-170288011600652 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1434403724.36-170288011600652 && echo $HOME/.ansible/tmp/ansible-tmp-1434403724.36-170288011600652
<aws-ansible> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<aws-ansible> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 aws-ansible rc=flag; [ -r /usr/src/python ] || rc=2; [ -f /usr/src/python ] || rc=1; [ -d /usr/src/python ] && rc=3; python -V 2>/dev/null || rc=4; [ x"$rc" != "xflag" ] && echo "${rc}  "/usr/src/python && exit 0; (python -c 'import hashlib; BLOCKSIZE = 65536; hasher = hashlib.sha1();
afile = open("'/usr/src/python'", "rb")
buf = afile.read(BLOCKSIZE)
while len(buf) > 0:
    hasher.update(buf)
    buf = afile.read(BLOCKSIZE)
afile.close()
print(hasher.hexdigest())' 2>/dev/null) || (python -c 'import sha; BLOCKSIZE = 65536; hasher = sha.sha();
afile = open("'/usr/src/python'", "rb")
buf = afile.read(BLOCKSIZE)
while len(buf) > 0:
    hasher.update(buf)
    buf = afile.read(BLOCKSIZE)
afile.close()
print(hasher.hexdigest())' 2>/dev/null) || (echo '0  '/usr/src/python)
<aws-ansible> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<aws-ansible> PUT /usr/src/python/Python-2.7.10.tar.xz TO /home/ec2-user/.ansible/tmp/ansible-tmp-1434403724.27-46406947970554/source
fatal: [aws-ansible]: FAILED! => {"msg": "ERROR! file or module does not exist: /usr/src/python/Python-2.7.10.tar.xz", "failed": true}

**Comment by jgrigonis** [comment]
It looks like it's ignoring the copy parameter, and trying to copy it from my local machine.

**Comment by bcoca** [comment]
yes, that is what it looks like

**Comment by jonhadfield** [comment]
I'm having the same issue and have tracked it down to the value of copy being returned as unicode, even for a boolean.
In /lib/ansible/plugins/action/unarchive.py it has:
`copy    = self._task.args.get('copy', True)`
If you specify the unarchive copy parameter as False, 'copy' is actually being set to u'False' and therefore evaluating as True.

**Comment by cortopy** [comment]
I'm getting the same problem with external links and getting the message that source does not exist

**Comment by bcoca** [comment]
the module properly made this a boolean in the spec, but sadly that does not affect the action plugin (how it does local file fetching). I added a boolean function and it should now work (fixed via 4d35d8bd31ffcba41e41351065233cdfd83d0599)
